ampache/content/contents/code/ampache.js

```diff
@@ -356,20 +356,30 @@ var AmpacheResolver = Tomahawk.extend(Tomahawk.Resolver, {
             for (var i = 0; i < songs.length; i++) {
                 var song = songs[i];
 
-                results.push({
+                trackObject = {
                     artist: this._decodeEntity(Tomahawk.valueForSubNode(song, "artist")),
                     album: this._decodeEntity(Tomahawk.valueForSubNode(song, "album")),
                     track: this._decodeEntity(Tomahawk.valueForSubNode(song, "title")),
                     albumpos: Tomahawk.valueForSubNode(song, "track"),
                     //result.year = 0;//valueForSubNode(song, "year");
                     source: this.settings.name,
-                    url: "ampache://track/" + song.getAttribute("id"),
-                    //mimetype: valueForSubNode(song, "mime"), //FIXME what's up here? it was there before :\
-                    //result.bitrate = valueForSubNode(song, "title");
+                    url: this._decodeEntity(Tomahawk.valueForSubNode(song, "url")),
                     size: Tomahawk.valueForSubNode(song, "size"),
                     duration: Tomahawk.valueForSubNode(song, "time"),
                     score: Tomahawk.valueForSubNode(song, "rating")
-                });
+                };
+
+                //Check if this stuff is present, and if so append it
+                bitrateValue = this._decodeEntity(Tomahawk.valueForSubNode(song, "bitrate"));
+                mimetypeValue = this._decodeEntity(Tomahawk.valueForSubNode(song, "mime"));
+                
+                if (typeof bitrateValue !== "undefined" && bitrateValue !== null && bitrateValue != "undefined") 
```

Comment:
I'm not exactly sure what it will return, but I always make sure to check for undefined type and null. I do mostly Coffeescript work and CoffeeScript has a ? operator that basically checks if the variable exists in the same fashion I did here. I had to add the "undefined" check because of the fact its a string. 
I suppose a micro performance boost would be to make the third check first.